pkg/controller/composite/controller.go

```diff
@@ -359,6 +366,20 @@ func (pc *parentController) updateParentObject(old, cur interface{}) {
 	// different status (e.g. you have some incrementing counter).
 	// Doing that is an anti-pattern anyway because status generation should be
 	// idempotent if nothing meaningful has actually changed in the system.
+	if parentOld, ok := old.(*unstructured.Unstructured); ok {
+		if parentCur, ok := cur.(*unstructured.Unstructured); ok {
+			// if ignoreStatusChanges is set to true in the decorator controller, a parent object should only be
+			// enqueues if either there is a change in the generation or if there is a change in its labels/annotations,
+			// otherwise it will be ignored.
+			if pc.parentIgnoreStatusChanges.Compares(parentCur) {
```

Comment:
not needed, why not inline to access 
```
if pc.cc.Spec.ParentResource.IgnoreStatusChanges
```
directly ?

We have controller created for specific `CR` so it is either enabled or disabled (or not set which is also disabled) which can be checked here